hkmc2/shared/src/test/mlscript/basics/BadMemberProjections.mls

```diff
@@ -10,27 +10,32 @@
 //│ ╟── Note: any expression of the form `‹expression›::‹identifier›` is a member projection;
 //│ ╙──   add a space before ‹identifier› to make it an operator application.
 //│ JS (unsanitized):
-//│ (self, ...args) => { return runtime.safeCall(self.x(...args)) }
-//│ = [function block$res1]
+//│ let lambda;
+//│ lambda = function lambda(self, ...args) {
+//│   return runtime.safeCall(self.x(...args))
+//│ };
+//│ lambda
+//│ = [function lambda]
 
 :ssjs
 :e
 :re
 1::x()
 //│ ╔══[ERROR] Integer literal is not a known class.
-//│ ║  l.19: 	1::x()
+//│ ║  l.23: 	1::x()
 //│ ║        	^
 //│ ╟── Note: any expression of the form `‹expression›::‹identifier›` is a member projection;
 //│ ╙──   add a space before ‹identifier› to make it an operator application.
 //│ JS:
-//│ block$res2 = runtime.checkCall(((...args1) => {
-//│   globalThis.Predef.checkArgs("", 1, false, args1.length);
+//│ lambda1 = function lambda(...args1) {
+//│   globalThis.Predef.checkArgs("lambda", 1, false, args1.length);
 //│   let self = args1[0];
 //│   let args = globalThis.Predef.tupleSlice(args1, 1, 0);
 //│   return runtime.safeCall(self.x(...args))
-//│ })());
+//│ };
+//│ block$res2 = runtime.checkCall(lambda1());
 //│ undefined
-//│ ═══[RUNTIME ERROR] Error: Function expected at least 1 argument but got 0
+//│ ═══[RUNTIME ERROR] Error: Function 'lambda' expected at least 1 argument but got 0
```

Comment:
It would be good if we could avoid making every lambda get the name "lambda", at least in such error messages. Ideally they'd be unnamed and we would still have the old message here.

Reply:
Currently it seems `BlockMemberSymbol`s must have names. Maybe we could add an flag which disables the name from being printed in error messages -- I don't know if it makes sense for BlockMemberSymbols to be unnamed.

Reply:
I'd say that if the symbol was not named by the user, it really shouldn't get a name like this. I checked and using `""` works, here (but then all lambdas are bound to `tmp`-like names, which is a pity).

We could also add a flag that says the name is just a temporary variable name hint and not provided by the user, with the same effect as the flag you proposed.